source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -417,9 +417,11 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
                                     !tls_certificate.password().empty()
                                         ? const_cast<char*>(tls_certificate.password().c_str())
                                         : nullptr));
+
         if (pkey == nullptr || !SSL_CTX_use_PrivateKey(ctx.ssl_ctx_.get(), pkey.get())) {
-          throw EnvoyException(
-              absl::StrCat("Failed to load private key from ", tls_certificate.privateKeyPath()));
+          throw EnvoyException(fmt::format("Failed to load private key from {}, Cause: {}",
+                                           tls_certificate.privateKeyPath(),
+                                           Utility::getLastCryptoError().value_or("not found")));
```

Comment:
"not found" here is misleading, since it sounds like file was not found.

Reply:
Yeah,  I think you're right.
Does **Unknown** sound sane to you?

Reply:
Yes, unknown is much better

Reply:
`"unknown"` sounds fine, but all other uses of `Utility::getLastCryptoError()` in this file use `.value_or("")`.

I'm fine with either.

Reply:
Cool, I updated it.
At the same time, I noticed that there're a few other places that could have improvements similar to this.
I'd like to help if you think it's nice to have them.

Reply:
Go for it. Thanks!